Jetchat/app/src/main/java/com/example/compose/jetchat/components/JetchatDrawer.kt

```diff
@@ -0,0 +1,171 @@
+import androidx.compose.foundation.layout.preferredWidth
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.AmbientContentAlpha
+import androidx.compose.material.AmbientContentColor
+import androidx.compose.material.ContentAlpha
+import androidx.compose.material.Divider
+import androidx.compose.material.Icon
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Providers
+import androidx.compose.ui.Alignment.Companion.CenterVertically
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.imageResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.example.compose.jetchat.R
+import com.example.compose.jetchat.data.colleagueProfile
+import com.example.compose.jetchat.data.meProfile
+import com.example.compose.jetchat.theme.JetchatTheme
+
+@Composable
+fun ColumnScope.JetchatDrawer(onProfileClicked: (String) -> Unit, onChatClicked: (String) -> Unit) {
+    DrawerHeader()
+    Divider()
+    DrawerItemHeader("Chats")
+    ChatItem("composers", true) { onChatClicked("composers") }
+    ChatItem("droidcon-nyc", false) { onChatClicked("droidcon-nyc") }
+    DrawerItemHeader("Recent Profiles")
+    ProfileItem("Ali Conors (you)", meProfile.photo) { onProfileClicked(meProfile.userId) }
+    ProfileItem("Taylor Brooks", colleagueProfile.photo) { onProfileClicked(colleagueProfile.userId) }
+}
+
+@Composable
+private fun DrawerHeader() {
+    Row(modifier = Modifier.padding(16.dp), verticalAlignment = CenterVertically) {
+        Image(
+            vectorResource(id = R.drawable.ic_jetchat),
+            modifier = Modifier.preferredSize(24.dp)
+        )
+        Image(
+            vectorResource(id = R.drawable.jetchat_logo),
+            modifier = Modifier.padding(start = 8.dp)
+        )
+    }
+}
+@Composable
+private fun DrawerItemHeader(text: String) {
+    Providers(AmbientContentAlpha provides ContentAlpha.medium) {
+        Text(text, style = MaterialTheme.typography.caption, modifier = Modifier.padding(16.dp))
+    }
+}
+
+@Composable
+private fun ChatItem(text: String, selected: Boolean, onChatClicked: () -> Unit) {
+    val background = if (selected) {
+        Modifier.background(MaterialTheme.colors.primary.copy(alpha = 0.08f))
+    } else {
+        Modifier
+    }
+    Row(
+        modifier = Modifier
+            .preferredHeight(48.dp)
+            .fillMaxWidth()
+            .padding(horizontal = 8.dp, vertical = 4.dp)
+            .then(background)
+            .clip(MaterialTheme.shapes.medium)
+            .clickable(onClick = onChatClicked),
+        verticalAlignment = CenterVertically
+    ) {
+        val mediumEmphasisOnSurface =
```

Comment:
This is only used if selected, maybe rename this to `iconTint` and have the tint logic inlined?

Reply:
I kept it not inlined but moved the logic around